### PR TITLE
Remove X-Pack centric graph endpoints

### PR DESCRIPTION
**Pull request by jasontedor**
This commit is part of our plan to deprecate and ultimately remove the use of _xpack in the REST APIs.

Relates #35958

### CODE REVIEW

File: x-pack/plugin/src/test/resources/rest-api-spec/api/xpack.graph.explore.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
     "documentation": "https://www.elastic.co/guide/en/elasticsearch/reference/current/graph-explore-api.html",
     "methods": ["GET", "POST"],
     "url": {
-      "path": "/{index}/_xpack/graph/_explore",
-      "paths": ["/{index}/_xpack/graph/_explore", "/{index}/{type}/_xpack/graph/_explore"],
+      "path": "/{index}/_xpack/_graph/explore",
+      "paths": ["/{index}/_xpack/_graph/explore", "/{index}/{type}/_xpack/_graph/explore"],
       "parts" : {
```
Comment (markharwood) [inline x-pack/plugin/src/test/resources/rest-api-spec/api/xpack.graph.explore.json]: Remove the `_xpack`?

Comment (jasontedor) [reply x-pack/plugin/src/test/resources/rest-api-spec/api/xpack.graph.explore.json]: @markharwood Oops, thanks! I pushed a2dc134a454d90b677580d0e185186d26eee239d.

**Comment by markharwood** [comment]
Do the yaml tests now need a "skip" marker for <7.0 versions now that we changed the endpoint? Presumably the BWC tests would fail if a 7.x client sends to a 6.x cluster?

**Comment by jasontedor** [comment]
@markharwood Yeah, I pushed 8118f0302de6a199645bc640da26c048c8f5633f.

**Comment by markharwood** [comment]
LGTM. Kibana Graph UI has a small label in the "last request" spy that shows the URL used for the graph call - I can change that.

**Pull request merged**